src/app/validators/company.async-validations.ts

```diff
@@ -1,53 +1,78 @@
+      group('company', () =>
+        companyAsyncValidations(model, field, groupName, context)
+      );
+    }
+  ) as ValidationSuite;
 }
 
-export const companyAsyncValidationSuite: ValidationSuite = createCompanyAsyncValidationSuite();
+export const companyAsyncValidationSuite: ValidationSuite =
+  createCompanyAsyncValidationSuite();
 
 /** Company Asynchronous Validation Suite */
-export const companyAsyncValidations: ValidationSuiteFn =
-  (model: Partial<Company>, field?: string, groupName?: string, context?: ValidationContext) => {
-    model = model ?? {};
+export const companyAsyncValidations: ValidationSuiteFn = (
+  model: Partial<Company>,
+  field?: string,
+  groupName?: string,
+  context?: ValidationContext
+) => {
+  model = model ?? {};
 
-    const fvs = (<AppValidationContext>context)?.feinValidationService;
-    /** FEIN check function (or FN returning "not found" result) */
-    const checkFein = fvs
-      ? fvs.check.bind(fvs)
-      // No fvs? This is almost certainly a bug. Remember to provide a context.
-      : (fein: string) => Promise.resolve<FeinValidationResponse>({ fein });
+  const fvs = (<AppValidationContext>context)?.feinValidationService;
+  /** FEIN check function (or FN returning "not found" result) */
+  const checkFein = fvs
+    ? fvs.check.bind(fvs)
+    : // No fvs? This is almost certainly a bug. Remember to provide a context.
+      (fein: string) => Promise.resolve<FeinValidationResponse>({ fein });
 
+  omitWhen(!isGoodFein(model.fein), () => {
+    // these validations will be completely omitted if the FEIN is not valid
 
     test('legalName', 'Legal Name must match Federal Tax Name', async () => {
-      if (isGoodFein(model.fein) && model.legalName) {
+      if (model.legalName) {
         const response = await checkFein(model.fein!);
         // console.log('legal name checkFein response', response);
         const name = response.feinName;
-        if (name && name !== '*' && name !== model.legalName.toUpperCase()) {
-          throw `Does not match "${name}", the name registered with the IRS for this tax number`;
-        }
+
+        enforce(name)
+          .message(
+            `Does not match "${name}", the name registered with the IRS for this tax number`
+          )
+          .isNotBlank()
+          .notEquals('*')
+          .notEquals(model.legalName.toUpperCase());
       }
-      return ''; // All other cases are either OK or the test does not apply
+      // All other cases are either OK or the test does not apply
     });
 
-
     test('fein', 'Tax Number is not registered with the IRS', async () => {
-      if (isGoodFein(model.fein)) {
-        const response = await checkFein(model.fein!);
-        // console.log('fein checkFein response', response);
-        enforce(response.feinName).isNotNullish();
-      }
-      return ''; // All other cases are either OK or the test does not apply
```

Comment:
This is not needed. Vest really ignores any return value other than:
- thrown Error
- thrown String
- `false` (literal false, not falsy values)
- Promise rejection

https://vestjs.dev/docs/writing_tests/using_the_test_function#how-to-fail-a-test